src/shims/intrinsics.rs

```diff
@@ -29,6 +27,22 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         // that might still hang around!
 
         let intrinsic_name = &*this.tcx.item_name(instance.def_id()).as_str();
+
+        // The 'abort' intrinsic has an empty 'dest', so we check for it before
+        // unwrapping it
+        if intrinsic_name == "abort" {
+            throw_ub_format!("the evaluated program aborted!");
+        }
+
+        if this.emulate_intrinsic(instance, args, dest)? {
+            return Ok(());
+        }
+
+        let dest = match dest {
+            Some(dest) => dest,
+            None => throw_ub_format!("Missing dest for intrinsic {}", intrinsic_name)
```

Comment:
Hm, I see where @oli-obk was going in the other thread. This should be an assert, it should never be user-reachable. `throw_ub` should indicate that this is actual UB a user has to worry about.